src/js/media-controller.js

```diff
@@ -36,6 +36,25 @@ class MediaController extends MediaContainer {
     // Track externally associated control elements
     this.associatedElements = [];
 
+    const mutationCallback = (mutationsList, _observer) => {
+      const { addedNodes, removedNodes } = toNextMediaChromeNodesState(mutationsList);
+      addedNodes.forEach(this.associateElement.bind(this));
+      removedNodes.forEach(this.unassociateElement.bind(this));
+    };
+
+    const observer = new MutationObserver(mutationCallback);
+    observer.observe(this, { childList: true, attributes: true, subtree: true });
+
+    /** option 2, controller side */
+    this.addEventListener(MediaUIEvents.MEDIA_CHROME_ELEMENT_CONNECTED, ({ target, path = [] }) => {
+      const el = path?.[0] ?? target;
```

Comment:
Use the root of the path if available, otherwise use target (using optional chaining and nullish coalescing)

The path will be the originating target/dispatcher, even on (non-closed) shadow DOMs,
 buuuuut it looks like the standards folks settled on `Event.composePath()` instead, so glad you called it out:
https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath